designer/test/data-model.test.js

```diff
@@ -0,0 +1,131 @@
+        { name: 'name3', page: { name: 'page2', section: 'section1' } },
+        { name: 'name4', page: { name: 'page2', section: 'section1' } }
+      ])
+    })
+
+    test('should ignore unnamed components', () => {
+      const data = new Data({
+        pages: [
+          {
+            name: 'page1',
+            section: 'section1',
+            components: [{ name: 'name1' }, { name: 'name2' }]
+          },
+          {
+            name: 'page2',
+            section: 'section1',
+            components: [{ badger: 'name3' }, { name: 'name4' }]
+          }
+        ]
+      })
+      expect(data.allInputs()).to.equal([
+        { name: 'name1', page: { name: 'page1', section: 'section1' } },
+        { name: 'name2', page: { name: 'page1', section: 'section1' } },
+        { name: 'name4', page: { name: 'page2', section: 'section1' } }
+      ])
+    })
+
+    test('should handle no pages', () => {
+      const data = new Data({ pages: [] })
+      expect(data.allInputs()).to.equal([])
+    })
+
+    test('should handle undefined pages', () => {
+      const data = new Data({ })
+      expect(data.allInputs()).to.equal([])
+    })
+
+    test('should handle pages with undefined components', () => {
+      const data = new Data({
+        pages: [{ }]
+      })
+      expect(data.allInputs()).to.equal([])
+    })
+
+    test('should handle pages with no components', () => {
+      const data = new Data({
+        pages: [{ components: [] }]
+      })
+      expect(data.allInputs()).to.equal([])
+    })
+  })
+
+  describe('find page', () => {
+    test('should return the page with the requested path if it exists', () => {
+      const data = new Data({
+        pages: [
+          {
+            name: 'page1',
+            section: 'section1',
+            path: '/1',
+            next: ['/2'],
+            components: [{ name: 'name1' }, { name: 'name2' }]
+          },
+          {
+            name: 'page2',
+            section: 'section1',
+            path: '/2',
+            next: ['/3'],
+            components: [{ name: 'name3' }, { name: 'name4' }]
+          }
+        ]
+      })
+      expect(data.findPage('/2')).to.equal(data.pages[1])
+    })
+
+    test('should return undefined if the requested page does not exist', () => {
+      const data = new Data({
+        pages: [
+          {
+            name: 'page1',
+            section: 'section1',
+            path: '/1',
+            next: ['/3'],
+            components: [{ name: 'name1' }, { name: 'name2' }]
+          }
+        ]
+      })
+
+      expect(data.findPage('/2')).to.equal(undefined)
+    })
+
+    test('should handle no pages', () => {
+      const data = new Data({ pages: [] })
+      expect(data.findPage('/1')).to.equal(undefined)
+    })
+
+    test('should handle undefined pages', () => {
+      const data = new Data({ })
+      expect(data.findPage('/1')).to.equal(undefined)
+    })
+  })
+})
```

Comment:
would it be worth testing a call from a mounted component?

Reply:
I'm not sure, as I'm not overly familiar with React and any oddities that might exist there. I thought this was just standard javascript so it wouldn't need testing in that way. Is there something I'm not aware of that would mean it's necessary? I can certainly add it if you thing required, I'm just a bit nervous of it as I suspect we'll want to move this to engine / a shared model library at some point

Reply:
I had this with the save method in mind, if it was persisting correctly and findPage returns with new data for example. If we leave data.save() for another PR then we can test that there

Reply:
cool, will do